File: java/tst/tarehart/rlbot/tuning/CarPathHelper.java
package tarehart.rlbot.tuning;

import com.google.gson.Gson;
import tarehart.rlbot.Bot;
import tarehart.rlbot.carpredict.AccelerationModel;
import tarehart.rlbot.carpredict.CarPath;
import tarehart.rlbot.carpredict.CarSlice;
import tarehart.rlbot.input.CarData;
import tarehart.rlbot.input.CarSpin;
import tarehart.rlbot.physics.DistancePlot;
import tarehart.rlbot.time.Duration;

import java.io.InputStream;
import java.util.Scanner;

public class CarPathHelper {

    public CarPath readRecording(String filename) {
        InputStream in = getClass().getResourceAsStream("/carrecordings/" + filename);
        Scanner s = new Scanner(in).useDelimiter("\\A");
        String content = s.hasNext() ? s.next() : "";

        Gson gson = new Gson();
        return gson.fromJson(content, CarPath.class);
    }

    public DistancePlot makePrediction(CarPath actual, boolean hasBoost) {
        Duration duration = Duration.Companion.between(actual.getSlices().get(0).getTime(), actual.getSlices().get(actual.getSlices().size() - 1).getTime());
        CarData carData = getCarDataFromSlice(actual.getSlices().get(0));
        return AccelerationModel.INSTANCE.simulateAcceleration(carData, duration, hasBoost ? 100 : 0, 0);
    }

    public CarData getCarDataFromSlice(CarSlice carSlice) {
        return new CarData(
                carSlice.getSpace(),
                carSlice.getVelocity(),
                carSlice.getOrientation(),
                new CarSpin(0, 0, 0),
                100,
                false,
                Bot.Team.BLUE,
                0,
                carSlice.getTime(),
                0,
                true,
                false,
                "");
    }

}

File: src/test/java/tarehart/rlbot/tuning/CarPathHelper.kt
package tarehart.rlbot.tuning

import com.google.gson.Gson
import rlbot.render.NamedRenderer
import tarehart.rlbot.bots.Team
import tarehart.rlbot.carpredict.AccelerationModel
import tarehart.rlbot.carpredict.CarPath
import tarehart.rlbot.carpredict.CarSlice
import tarehart.rlbot.input.CarData
import tarehart.rlbot.input.CarHitbox
import tarehart.rlbot.input.CarSpin
import tarehart.rlbot.math.vector.Vector3
import tarehart.rlbot.physics.DistancePlot
import tarehart.rlbot.time.Duration
import java.util.*

class CarPathHelper {

    fun readRecording(filename: String): CarPath {
        val streamIn = javaClass.getResourceAsStream("/tst/carrecordings/$filename")
        val s = Scanner(streamIn).useDelimiter("\\A")
        val content = if (s.hasNext()) s.next() else ""

        val gson = Gson()
        return gson.fromJson(content, CarPath::class.java)
    }

    fun makePrediction(actual: CarPath, hasBoost: Boolean): DistancePlot {
        val duration = Duration.between(actual.firstSlice.time, actual.lastSlice.time)
        val carData = getCarDataFromSlice(actual.firstSlice)
        return AccelerationModel.simulateAcceleration(carData, duration, (if (hasBoost) 100 else 0).toFloat(), 0.0)
    }

    private fun getCarDataFromSlice(carSlice: CarSlice): CarData {
        return CarData(
                carSlice.space,
                carSlice.velocity,
                carSlice.orientation,
                CarSpin(0F, 0F, 0F, Vector3()),
                100F,
                false,
                Team.BLUE,
                0,
                carSlice.time,
                0,
                true,
                false,
                "",
                NamedRenderer("test"),
                isBot = true,
                hitbox = CarHitbox.OCTANE)
    }

}
